src/pilot/assign_role.py

```diff
@@ -309,11 +313,15 @@ def define_compute_logical_disks(drac_client, raid_controller_name):
     raid_10_logical_disk = define_single_raid_10_logical_disk(
         drac_client, raid_controller_name)
 
+    # None indicates an error occurred.
     if raid_10_logical_disk is None:
         return None
 
     logical_disks = list()
-    logical_disks.append(raid_10_logical_disk)
+
+    # Add the disk to the list only if it is not empty.
+    if logical_disks:
+        logical_disks.append(raid_10_logical_disk)
```

Comment:
Same as above, logical_disks is always empty as it is initialized on line #320 above

Reply:
Changed.